### PR TITLE
handle autofix flag in payload

### PR DESCRIPTION
**Pull request by jennygzma**
define INPUT_AUTOFIX if inputs.check-mode === payload.

if INPUT_AUTOFIX is defined, then run trunk check and push results to the github branch

### CODE REVIEW

File: pull_request.sh [hunk 1]
```diff
@@ -48,10 +48,16 @@ else
...
 
-"${TRUNK_PATH}" check \
-  --ci \
-  --upstream "${upstream}" \
-  --github-commit "${git_commit}" \
-  --github-label "${INPUT_LABEL}" \
-  "${annotation_argument}" \
-  ${INPUT_ARGUMENTS}
+if [[ -n ${INPUT_AUTOFIX} ]]; then
+  "${TRUNK_PATH}" check --fix
```
Comment (sxlijin) [inline pull_request.sh:52]:
```suggestion
  "${TRUNK_PATH}" check --ci --upstream "${upstream}" --fix
```

**Review by sxlijin — COMMENTED** [review]
add tests like what we've done for trunk_merge and all-hold-the-line-new-series

File: action.yaml [hunk 2]
```diff
@@ -161,6 +161,7 @@ runs:
         INPUT_UPLOAD_LANDING_STATE=$(payload uploadLandingState)
         INPUT_UPLOAD_SERIES=$(payload uploadSeries)
         INPUT_UPLOAD_ID=$(payload uploadId)
+        INPUT_AUTOFIX=$(payload autofix)
```
Comment (sxlijin) [inline action.yaml:164]: Let's make this naming a bit more obvious

```suggestion
        INPUT_AUTOFIX_AND_PUSH=$(payload autofixAndPush)
```

File: .github/workflows/action_tests.yaml [hunk 3]
```diff
@@ -145,3 +145,44 @@ jobs:
...
+
+      - name: Set up test
+        shell: bash
+        run: ./local-action/action_tests/setup.sh src-repo repo-under-test
+
+      - name: Run trunk-action
+        shell: bash
+        run: |
+          cd repo-under-test
+          git checkout trunk-merge/of-feature-branch
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:165]:
```suggestion
          git checkout feature-branch
```

File: pull_request.sh [hunk 4]
```diff
@@ -48,10 +48,24 @@ else
...
-  --upstream "${upstream}" \
-  --github-commit "${git_commit}" \
-  --github-label "${INPUT_LABEL}" \
-  "${annotation_argument}" \
-  ${INPUT_ARGUMENTS}
+github_path=git
+# if in test, we need to fake out calling git too
+if [[ ${TRUNK_PATH} == "../local-action/action_tests/stub.js" ]]; then
+  github_path=${TRUNK_PATH}
+fi
```
Comment (sxlijin) [inline pull_request.sh:55]: Hmm. I hear what you're saying, and I like the idea, but I don't think we need to do this. Rather, we should be able to, in assert.js, if we're doing the `pull-request-autofix` test case, make assertions about the state of `feature-branch` and `origin/feature-branch`

Comment (jennygzma) [reply pull_request.sh:55]: i think the problem is i can't push an empty commit into github - id have to fake a file change and push up a commit which im not super sure how to do

Comment (sxlijin) [reply pull_request.sh:55]: but you're not pushing to github - `repo-under-test` is set to push to `src-repo`

**Comment by jennygzma** [comment]
closing - too messy

**Pull request closed without merging**